### PR TITLE
Move big archive loading into a seperate project

### PR DESCRIPTION
**Pull request by feliwir**
Moved the big loading code into `OpenSage.FileFormats.Big`. Also removed the dependency of the BigEditor on `OpenSage.Game` by introducing `OpenSage.Core`, which contains game unrelated classes.

### CODE REVIEW

File: src/OpenSage.FileFormats.Big/BigArchive.cs [hunk 1]
```diff
@@ -4,13 +4,11 @@
...
 using System.Threading;
-using OpenSage.Data.Utilities.Extensions;
-using OpenSage.FileFormats;
 
-namespace OpenSage.Data.Big
+namespace OpenSage.FileFormats.Big
 {
     [DebuggerDisplay("Archive: {FilePath}")]
-    public class BigArchive : DisposableBase
+    public class BigArchive : IDisposable
```
Comment (tgjones) [inline src/OpenSage.FileFormats.Big/BigArchive.cs]: Why change this? Or did you make this change before you moved DisposableBase to OpenSage.Core...

File: src/OpenSage.FileFormats.Refpack/OpenSage.FileFormats.RefPack.csproj [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (tgjones) [inline src/OpenSage.FileFormats.Refpack/OpenSage.FileFormats.RefPack.csproj]: Sorry to be a pain, but please could you use consistent casing for the project (and folder) name? RefPack, not Refpack.

File: src/OpenSage.FileFormats.Big/BigArchive.cs (hunk 1 above)
Comment (feliwir) [reply src/OpenSage.FileFormats.Big/BigArchive.cs]: Yes, that was done before. Will revert this

File: src/OpenSage.FileFormats.Refpack/OpenSage.FileFormats.RefPack.csproj (hunk 2 above)
Comment (tgjones) [reply src/OpenSage.FileFormats.Refpack/OpenSage.FileFormats.RefPack.csproj]: Now there seem to be duplicate files, under both Refpack and RefPack...

Comment (feliwir) [reply src/OpenSage.FileFormats.Refpack/OpenSage.FileFormats.RefPack.csproj]: Case insensitivity on windows is really disgusting....

Comment (feliwir) [reply src/OpenSage.FileFormats.Refpack/OpenSage.FileFormats.RefPack.csproj]: This seems to be correct for me. There are no multiple files on git

**Pull request merged**